packages/runtime/container-runtime/src/gc/garbageCollection.ts

```diff
@@ -856,21 +864,35 @@ export class GarbageCollector implements IGarbageCollector {
 	 * @param local - Whether it was send by this client.
 	 */
 	public processMessage(message: ContainerRuntimeGCMessage, local: boolean) {
-		switch (message.contents.type) {
-			case "Sweep": {
+		const gcMessageType = message.contents.type;
+		switch (gcMessageType) {
+			case GarbageCollectionMessageType.Sweep: {
 				// Delete the nodes whose ids are present in the contents.
 				this.deleteSweepReadyNodes(message.contents.deletedNodeIds);
 				break;
 			}
+			case GarbageCollectionMessageType.TombstoneLoaded: {
+				if (this.mc.config.getBoolean(disableAutoRecoveryKey) === true) {
+					break;
+				}
+
+				//* NOTE: In separate PR... Then also run full GC - Otherwise if it's our bug it won't resolve (the object with the handle we missed doesn't change)
+
+				// Mark the node as referenced to ensure it isn't Swept
+				const tombstonedNodePath = message.contents.nodePath;
+				this.addedOutboundReference("/", tombstonedNodePath);
```

Comment:
Thinking about adding a parameter here to be logged to indicate it's due to this op. Thoughts?

Reply:
Should we add a separate log that indicates we did auto-recovery for this object. Then we can separately track that in telemetry and know how often it happens.

Reply:
I think it's intuitive to call Auto-Recovery a subset of "Revived", in terms of how it's logged. I'll add a flag to that log.

We could also log on triggerAutoRecovery

Reply:
We don't need a log when triggering autorecovery because it's right next to the GC_Tombstone log.